product_multi_price/models/product_product.py

```diff
@@ -15,9 +15,7 @@ class ProductProduct(models.Model):
     def _get_multiprice_pricelist_price(self, rule):
         """Method for getting the price from multi price."""
         self.ensure_one()
-        company = rule.company_id or self.env.user.company_id
-        price = self.env['product.multi.price'].sudo().search([
-            ('company_id', '=', company.id),
```

Comment:
This was needed for calls to this in sudo mode (which happens in some places, for example, in e-commerce).